backend/src/main/java/com/redhat/cloud/notifications/routers/EndpointResource.java

```diff
@@ -337,10 +339,98 @@ public Response updateEndpoint(@Context SecurityContext sec, @PathParam("id") UU
+                String url;
+
+            };
+            if (ep != null) {
+                EndpointProperties properties = ep.getProperties();
+                if (properties instanceof CamelProperties) {
+                    CamelProperties cp = (CamelProperties) properties;
+
+                    ref.processorId = cp.getExtras().get(OB_PROCESSOR_ID);
+                    ref.processorName = cp.getExtras().get(OB_PROCESSOR_NAME);
+                    CamelProperties out = endpoint.getProperties(CamelProperties.class);
+                    out.getExtras().put(OB_PROCESSOR_NAME, ref.processorName);
+                    out.getExtras().put(OB_PROCESSOR_ID, ref.processorId);
+                    ref.channel = out.getExtras().get("channel");
+                    ref.url = cp.getUrl();
+                }
+            }
+
+            try {
+                UnaryOperator<Processor> updateFun = in -> {
+                    in.getAction().getParameters().put(SLACK_CHANNEL, ref.channel);
+                    in.getAction().getParameters().put(SLACK_WEBHOOK_URL, ref.url);
+                    in.setTransformationTemplate(in.getTransformationTemplate() + " ---- updated"); // TODO remove
+                    ep.setStatus(EndpointStatus.PROVISIONING);
+                    return in;
+                };
+                UnaryOperator<Processor> execFun = in -> bridgeApiService.updateProcessor(bridge.getId(), in.getId(), bridgeAuth.getToken(), in);
+                Processor out = executeObAction(ep, updateFun, execFun);
+                Log.infof("Endpoint updated to processor %s ", out.toString());
+
+            } catch (Exception ex) {
+                Log.error("Update of processor failed: ", ex);
+                return Response.serverError().build();
+            }
+        }
+
         endpointRepository.updateEndpoint(endpoint);
+
         return Response.ok().build();
     }
 
+    /**
+     * Helper that has some boilerplate code when dealing with OpenBridge, which
+     * then in turn calls the provided functions to perform some work.
+     * @param endpoint Endpoint definition to operate on
+     * @param updateFunc A function that modifies the data structure of a created processor. Can be null
+     * @param executeFunc A function that talks with the remote to perform an action
+     * @return
+     */
+    Processor executeObAction(Endpoint endpoint, UnaryOperator<Processor> updateFunc, UnaryOperator<Processor> executeFunc) {
+        if (endpoint != null) {
+            EndpointProperties properties = endpoint.getProperties();
+            if (properties != null && properties instanceof CamelProperties) {
+                CamelProperties cp = (CamelProperties) properties;
+                // Special case wrt OpenBridge
+                if (endpoint.getSubType().equals(SLACK)) {
+                    String processorId = cp.getExtras().get(OB_PROCESSOR_ID);
+                    Processor p = createProcessor(endpoint);
```

Comment:
Could we avoid creating a processor here when this method is called to delete the processor on the OpenBridge side? The only data that is used in that case is the `processorId`, everything else is unused.

Reply:
I don't see deletes happen too often, so the memory/garbage collection overhead should be low. 
Or did you mean something else?

Reply:
I'm more worried by the code complexity than the memory/GC 😃.